controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -0,0 +1,238 @@
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Graph struct {
+	Title string `json:"Title"`
+	Query string `json:"Query"`
+}
+type MetricsDashboardConfigMapData struct {
+	Graphs []Graph `json:"graphs"`
+}
+
+var _ SubResourceReconciler = (*KserveMetricsDashboardReconciler)(nil)
+var ovmsData []byte
+var tgisData []byte
+var vllmData []byte
+
+type KserveMetricsDashboardReconciler struct {
+	NoResourceRemoval
+	client           client.Client
+	telemetryHandler resources.ConfigMapHandler
+	deltaProcessor   processors.DeltaProcessor
+}
+
+func NewKserveMetricsDashboardReconciler(client client.Client) *KserveMetricsDashboardReconciler {
+	return &KserveMetricsDashboardReconciler{
+		client:           client,
+		telemetryHandler: resources.NewConfigMapHandler(client),
+		deltaProcessor:   processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveMetricsDashboardReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// func (r *KserveMetricsDashboardReconciler) createDesiredResource(log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+
+	// resolve SR
+	inferenceServiceList := &kservev1beta1.InferenceServiceList{}
+	if err := r.client.List(ctx, inferenceServiceList, client.InNamespace(isvc.Namespace)); err != nil {
+		return nil, err
+	}
+
+	//create empty list to fill with all runtimes in the cluster
+	isvcServingRuntimeList := &kservev1alpha1.ServingRuntimeList{}
+
+	// go through each inference service to all runtimes
+	for i := len(inferenceServiceList.Items) - 1; i >= 0; i-- {
+		inferenceService := inferenceServiceList.Items[i]
+
+		// as long as the runtime field of isvc is not empty, find the runtime with that name and save it in a list
+		if inferenceService.Spec.Predictor.Model.Runtime != nil {
+			err := r.client.Get(ctx, types.NamespacedName{
+				Name:      *inferenceService.Spec.Predictor.Model.Runtime,
+				Namespace: isvc.Namespace,
+			}, isvcServingRuntimeList.Items[i])
+
+			// if there is error with getting the runtimes, throw an error
+			if err != nil {
+				if apierrs.IsNotFound(err) {
+					return nil, err
+				}
+			}
+	}
```

Comment:
each `InferenceService` will trigger the reconcile loop separately, so we don't need to fetch the list. So within the reconcile loop, include logic that you want to be executed for that 1 isvc. 

So here we can get the `ServingRuntime` for the `InferenceService` _that triggered this loop_ as follows : 
```suggestion
        isvcRuntime := isvc.Spec.Predictor.Model.Runtime
	runtime := &kservev1alpha1.ServingRuntime{}
	if err := r.client.Get(ctx, types.NamespacedName{Name: *isvcRuntime, Namespace: isvc.Namespace}, runtime); err != nil {
		log.Error(err, "Could not determine servingruntime for isvc")
	}
```